### PR TITLE
Add Table Transformer Detection QNN example

### PR DESCRIPTION
**Pull request by ssss141414**
## Describe your changes
Add Table Transformer Detection (https://huggingface.co/microsoft/table-transformer-detection) QNN example.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [x] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

## Metrics
Input model on CPU:
{
  "accuracy-mean_of_max_iou": 0.8218905329704285,
  "accuracy-f1_score": 0.9561123496676195,
  "latency-avg": 232.75268
}

output model on NPU:
{
    "accuracy-mean_of_max_iou": 0.8153600096702576,
    "accuracy-f1_score": 0.9516611285450918,
    "latency-avg": 124.70602,   
}

### CODE REVIEW

**Review by devang-ml — COMMENTED** [review]
Please add an entry in https://github.com/microsoft/Olive/blob/main/docs/source/examples.md for this model. Thanks!

File: examples/table_transformer_detection/ttd_config.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            ]
+        }
+    },
+    "passes": {
+        "conversion": { "device": "cpu", "type": "OnnxConversion", "target_opset": 17 },
+        "QNNPreprocess": { "type": "QNNPreprocess", "fuse_layernorm": true },
+        "OnnxQuantization": {
+            "type": "OnnxQuantization",
+            "data_config": "tablebank_data_config",
+            "op_types_to_quantize": [ "MatMul", "LayerNormalization", "Gemm", "Sigmoid", "Gelu" ],
```
Comment (jambayk) [inline examples/table_transformer_detection/ttd_config.json:51]: what are the ops being excluded? If it is add and softmax could you try this graph surgery to see if it fixes accuracy issues https://github.com/microsoft/Olive/blob/db8963b12d21e5558e09070a9d3ea8951b92e765/examples/bge/bge-small-en-v1.5_ptq_qnn.json#L64

If not, I would recommend using the new option `op_types_to_exclude` instead of this https://github.com/microsoft/Olive/blob/db8963b12d21e5558e09070a9d3ea8951b92e765/examples/adetailer/face_yolo_qnn.json#L101

File: docs/source/examples.md [hunk 2]
```diff
@@ -22,3 +22,5 @@
 ||VGG|[Link](https://github.com/microsoft/Olive/tree/main/examples/vgg)|`Qualcomm NPU`: with SNPE toolkit
 ||inception|[Link](https://github.com/microsoft/Olive/tree/main/examples/inception)|`Qualcomm NPU`: with SNPE toolkit
 ||super resolution|[Link](https://github.com/microsoft/Olive/tree/main/examples/super_resolution)|`CPU`: with ONNX Runtime pre/post processing integration for a single ONNX model
+||Vision Transformer|[Link](https://github.com/microsoft/Olive/tree/main/examples/vit)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
+||Table Transformer Detection|[Link](https://github.com/microsoft/Olive/tree/main/examples/table_transformer_detection)|`Qualcomm NPU`: with ONNX Runtime static QDQ quantization for ONNX Runtime QNN EP
```
Comment (jambayk) [inline docs/source/examples.md:26]: Please add the entry for this example in a follow up PR. Since this link does not exist on main yet, it will fail the doc build job.

File: examples/table_transformer_detection/ttd_config.json (hunk 1 above)
Comment (ssss141414) [reply examples/table_transformer_detection/ttd_config.json]: This model is fine-tuned DERT model. I think DERT contains attention layers... Also I can find like 'model.decoder.layers.0.encoder_attn.k_proj' on Netron.

It seems somehow in this case attention mask surgery affect not too much on the model...

**Review by jambayk — APPROVED** [review]
